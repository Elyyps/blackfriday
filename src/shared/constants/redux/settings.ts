export const SETTINGS = {
  SET_SCREEN_SIZE: "SET_SCREEN_SIZE",
  SET_BLACK_FRIDAY_DATE: "SET_BLACK_FRIDAY_DATE",
  SET_BLACK_FRIDAY_ROOT_URL: "SET_BLACK_FRIDAY_ROOT_URL",
  SET_DISTANCE_TOP: "SET_DISTANCE_TOP"
};
