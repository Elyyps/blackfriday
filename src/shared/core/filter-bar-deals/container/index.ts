export * from "./filter-bar-deals.container";
