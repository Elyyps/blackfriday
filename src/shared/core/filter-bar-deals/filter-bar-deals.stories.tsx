import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
// import { FilterBarDeals } from "./filter-bar-deals.component";
// import { ViewType } from "@app/stores/settings";
/* tslint:disable */
storiesOf("Filter bar", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <div />
    // <FilterBarDeals
    //   filtersChanged={() => {}}
    //   brandFilterItems={[]}
    //   categoryFilterItems={[]}
    //   screenSize={{ breakpointPixels: 0, viewType: ViewType.Desktop }}
    //   sortBy=""
    //   statusFilterItems={[]}
    //   totalResults={100}
    //   clearFilters={() => {}}
    //   setBrandFilters={() => {}}
    //   setCategoryFilters={() => {}}
    //   setSortBy={() => {}}
    //   setStoreFilters={() => {}}
    // />
  ));
