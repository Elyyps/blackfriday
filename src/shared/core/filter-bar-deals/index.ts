export * from "./filter-bar-deals.component";
export * from "./container";
