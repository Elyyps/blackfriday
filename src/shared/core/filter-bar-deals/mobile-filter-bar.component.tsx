import React, { useState, useEffect } from "react";

import { MobileFilterComponent, IMobileFilterSelectedItems } from "../mobile-filter";
import { FilterItem } from "@app/api/core/filter/filter-item";
import { IMobileFilterItem } from "../mobile-filter/mobile-filter-item";

export interface IDealsMobileFilterBarComponentProps {
  brandFilterItems: FilterItem[];
  categoryFilterItems: FilterItem[];

  onFiltersChanged: (
    brandFilterItems: FilterItem[],
    categoryFilterItems: FilterItem[],
    storeFilterItem: FilterItem[],
    sortBy: string
  ) => void;
  sortBy: string;
  sortByOptions: string[];
  storeFilterItems: FilterItem[];
}

const totalStores = 10; // TODO: calculate filter totals!
const DealsMobileFilterBarComponent = (props: IDealsMobileFilterBarComponentProps) => {
  const [mobileFilters, setMobileFilters] = useState<IMobileFilterItem[]>([]);

  useEffect(() => {
    setMobileFilters(
      getMobileFilterItems(
        props.brandFilterItems,
        props.categoryFilterItems,
        props.sortBy,
        props.sortByOptions,
        props.storeFilterItems
      )
    );
  }, [props.brandFilterItems, props.categoryFilterItems, props.sortBy, props.sortByOptions, props.storeFilterItems]);

  const handleFinishSearch = (selectedItems: IMobileFilterSelectedItems[]) => {
    const brandFilters = props.brandFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });
    const categoryFilters = props.categoryFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });
    const storeFilters = props.storeFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });

    let sortBy = "";

    selectedItems.map(item => {
      if (item.title === "Merk") {
        brandFilters.forEach((filter: FilterItem) => {
          if (item.selectedItems.includes(filter.displayName)) {
            filter.isSelected = true;
          }
        });
      }
      if (item.title === "Categorieen") {
        categoryFilters.forEach((filter: FilterItem) => {
          if (item.selectedItems.includes(filter.displayName)) {
            filter.isSelected = true;
          }
        });
      }
      if (item.title === "Winkels") {
        storeFilters.forEach((filter: FilterItem) => {
          if (item.selectedItems.includes(filter.displayName)) {
            filter.isSelected = true;
          }
        });
      }
      if (item.title === "Sorteren") {
        sortBy = item.selectedItems.length > 0 ? item.selectedItems[0] : "Relevatie";
      }
    });

    props.onFiltersChanged(brandFilters, categoryFilters, storeFilters, sortBy);
  };

  const clearAllFilters = () => {
    const clearedBrandFilters = props.brandFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });
    const clearedCategoryFilters = props.categoryFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });
    const clearedStatusFilters = props.storeFilterItems.map((filter: FilterItem) => {
      const result = { ...filter, isSelected: false };

      return result;
    });

    props.onFiltersChanged(clearedBrandFilters, clearedCategoryFilters, clearedStatusFilters, "Relevantie");
  };

  return (
    <MobileFilterComponent
      onFinish={handleFinishSearch}
      totalStores={totalStores}
      onClear={clearAllFilters}
      filterItems={mobileFilters}
    />
  );
};

const getMobileFilterItems = (
  brandfilterItems: FilterItem[],
  categoryFilterItems: FilterItem[],
  selectedSortBy: string,
  sortByOptions: string[],
  storeFilterItems: FilterItem[]
): IMobileFilterItem[] => [
  {
    hasSearchBar: false,
    title: "Sorteren",
    searchBarPlaceholder: "Sorteren zoeken",
    selectedItems: [selectedSortBy],
    isSingleSelection: true,
    items: sortByOptions
  },
  {
    hasSearchBar: true,
    title: "Status",
    searchBarPlaceholder: "Status zoeken",
    selectedItems: [...getSelectedFilterItems(storeFilterItems)],
    items: [...getFilterItems(storeFilterItems)]
  },
  {
    hasSearchBar: true,
    title: "Categorieen",
    searchBarPlaceholder: "Categorieen zoeken",
    selectedItems: [...getSelectedFilterItems(categoryFilterItems)],
    items: [...getFilterItems(categoryFilterItems)]
  },
  {
    hasSearchBar: true,
    title: "Merk",
    searchBarPlaceholder: "Merk zoeken",
    selectedItems: [...getSelectedFilterItems(brandfilterItems)],
    items: [...getFilterItems(brandfilterItems)]
  }
];

const getFilterItems = (items: FilterItem[]): string[] => items.map(item => item.displayName);

const getSelectedFilterItems = (items: FilterItem[]): string[] => {
  const selectedItems = items.filter(item => item.isSelected);

  return selectedItems.map(item => item.displayName);
};

export { DealsMobileFilterBarComponent };
