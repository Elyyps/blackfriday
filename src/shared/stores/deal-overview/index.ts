export * from "./deal-overview.action";
export * from "./deal-overview.reducer";
export * from "./deal-overview.types";
