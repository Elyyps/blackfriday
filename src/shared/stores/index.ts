export * from "./page";
export * from "./store";
export * from "./history";
