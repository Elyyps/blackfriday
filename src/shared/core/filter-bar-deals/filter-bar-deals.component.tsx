import React from "react";
import styles from "./filter-bar-deals-component.module.scss";
import HandPointing from "@assets/icons/hand-pointing.svg";
import DealIcon from "@assets/icons/medal.svg";
import { IconComponent } from "@app/core";
import { FilterItem } from "@app/api/core/filter/filter-item";
import { SearchableCheckboxDropdown } from "../searchable-checkbox-dropdown/searchable-checkbox-dropdown.component";
import { SelectComponent } from "../select";
import { FilterBarDealsContainerProps } from "./container";
import { ViewType } from "@app/stores/settings";
import { DealsMobileFilterBarComponent } from "./mobile-filter-bar.component";
import { injectIntl, InjectedIntlProps } from "react-intl";

export interface IFilterBarDealsProps {
  filtersChanged: () => void;
}

const component = (props: IFilterBarDealsProps & FilterBarDealsContainerProps & InjectedIntlProps) => {
  const onStoreFilterItemsChanged = (items: FilterItem[]) => {
    if (getNumberOfSelectedfilters(items) !== getNumberOfSelectedfilters(props.storeFilterItems)) {
      props.setStoreFilters([...items]);
      props.filtersChanged();
    }
  };

  const onCategoryFilterItemsChanged = (items: FilterItem[]) => {
    if (getNumberOfSelectedfilters(items) !== getNumberOfSelectedfilters(props.categoryFilterItems)) {
      props.setCategoryFilters([...items]);
      props.filtersChanged();
    }
  };

  const onBrandFilterItemsChanged = (items: FilterItem[]) => {
    if (getNumberOfSelectedfilters(items) !== getNumberOfSelectedfilters(props.brandFilterItems)) {
      props.setBrandFilters([...items]);
      props.filtersChanged();
    }
  };

  const setSort = (sortByString: string) => {
    props.setSortBy(sortByString);
  };

  const sortByOptions = ["Relevatie", "Nieuwste", "Populair", "Deals A - Z", "Deals Z - A"];

  const getTotalNumberOfFilters = (): number =>
    props.storeFilterItems.filter(item => item.isSelected).length +
    props.categoryFilterItems.filter(item => item.isSelected).length +
    props.brandFilterItems.filter(item => item.isSelected).length;

  const mobileFiltersChanged = (
    brandFilterItems: FilterItem[],
    categoryFilterItems: FilterItem[],
    storeFilterItem: FilterItem[],
    sortBy: string
  ) => {
    props.setBrandFilters([...brandFilterItems]);
    props.setCategoryFilters([...categoryFilterItems]);
    props.setStoreFilters([...storeFilterItem]);
    props.setSortBy(sortBy);
  };

  const getNumberOfSelectedfilters = (filterItems: FilterItem[]) => filterItems.filter(item => item.isSelected).length;

  return (
    <div className={styles["filter-bar"]}>
      {props.screenSize && props.screenSize.viewType > ViewType.Tablet ? (
        <div className={`${styles["filter-bar__filter-container"]} ${"uk-container"}`}>
          <div className={`${styles["filter-bar__filter-list"]}`}>
            <div className={`${styles["filter-label"]}`}>
              {props.intl.formatMessage({ id: "filter-bar-label" })}
              <span>
                <IconComponent icon={HandPointing} size={"20px"} />
              </span>
            </div>

            <div>
              <SearchableCheckboxDropdown
                searchPlaceholder={props.intl.formatMessage(
                  { id: "filter-bar-search-placeholder" },
                  {
                    title: props.intl.formatMessage({ id: "deals-filter-bar-stores-filter-title" })
                  }
                )}
                deleteFilterLabel={props.intl.formatMessage(
                  { id: "filter-control-clear" },
                  {
                    title: props.intl.formatMessage({ id: "deals-filter-bar-stores-filter-title" })
                  }
                )}
                title={props.intl.formatMessage({ id: "deals-filter-bar-stores-filter-title" })}
                showFilterName={props.intl.formatMessage({ id: "deals-filter-bar-stores-filter-title" })}
                items={[...props.storeFilterItems.map(item => ({ ...item }))]}
                onChange={onStoreFilterItemsChanged}
              />
            </div>
            <div>
              <SearchableCheckboxDropdown
                searchPlaceholder={props.intl.formatMessage(
                  { id: "filter-bar-search-placeholder" },
                  {
                    title: props.intl.formatMessage({ id: "deals-filter-bar-brands-filter-title" })
                  }
                )}
                deleteFilterLabel={props.intl.formatMessage(
                  { id: "filter-control-clear" },
                  { title: props.intl.formatMessage({ id: "deals-filter-bar-brands-filter-title" }) }
                )}
                title={props.intl.formatMessage({ id: "deals-filter-bar-brands-filter-title" })}
                showFilterName={props.intl.formatMessage({ id: "deals-filter-bar-brands-filter-title" })}
                items={[...props.brandFilterItems.map(item => ({ ...item }))]}
                onChange={onBrandFilterItemsChanged}
              />
            </div>
            <div>
              <SearchableCheckboxDropdown
                searchPlaceholder={props.intl.formatMessage(
                  { id: "filter-bar-search-placeholder" },
                  {
                    title: props.intl.formatMessage({ id: "deals-filter-bar-category-filter-title" })
                  }
                )}
                deleteFilterLabel={props.intl.formatMessage(
                  { id: "filter-control-clear" },
                  {
                    title: props.intl.formatMessage({ id: "deals-filter-bar-category-filter-title" })
                  }
                )}
                title={props.intl.formatMessage({ id: "deals-filter-bar-category-filter-title" })}
                showFilterName={props.intl.formatMessage({ id: "deals-filter-bar-category-filter-title" })}
                items={[...props.categoryFilterItems.map(item => ({ ...item }))]}
                onChange={onCategoryFilterItemsChanged}
              />
            </div>
            {getTotalNumberOfFilters() > 0 && (
              <div className={styles["filter-bar__clear-filter"]}>
                <span
                  role="link"
                  onClick={() => {
                    props.clearFilters();
                    props.filtersChanged();
                  }}
                  style={{ color: "red", cursor: "pointer" }}
                >
                  {props.intl.formatMessage({ id: "filter-bar-clear-filters" })}({getTotalNumberOfFilters()})
                </span>
              </div>
            )}
          </div>

          <div className={styles["filter-bar__sort"]}>
            <div className={styles["filter__sort-filter"]}>
              <span className={styles["amount-of-shops"]}>
                <IconComponent icon={DealIcon} size={"20px"} />
              </span>
              {props.totalResults} {props.intl.formatMessage({ id: "tab-winkels" })}
            </div>
            <div className={styles["filter__sort-item"]}>
              {/* className={styles["filter__sort-change"]} */}
              {/* {orderBy} */}
              <SelectComponent
                options={["Relevatie", "Nieuwste", "Populair", "Winkels A - Z", "Winkels Z - A"]}
                onSelect={setSort}
              />
            </div>
          </div>
        </div>
      ) : (
        <DealsMobileFilterBarComponent
          brandFilterItems={props.brandFilterItems}
          categoryFilterItems={props.categoryFilterItems}
          sortBy={props.sortBy}
          sortByOptions={sortByOptions}
          storeFilterItems={props.storeFilterItems}
          onFiltersChanged={mobileFiltersChanged}
        />
      )}
    </div>
  );
};
const FilterBarDeals = injectIntl(component);
export { FilterBarDeals };
