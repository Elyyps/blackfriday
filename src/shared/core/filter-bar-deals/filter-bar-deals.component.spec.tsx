import React from "react";
import { ViewType } from "@app/stores/settings";
import { shallowWithIntl } from "enzyme-react-intl";
import { FilterBarDeals } from "./filter-bar-deals.component";
/* tslint:disable */
describe("[FilterBarDeals]", () => {
  it("should render without crashing", () => {
    const renderedComponent = shallowWithIntl(
      <FilterBarDeals
        filtersChanged={jest.fn()}
        brandFilterItems={[]}
        categoryFilterItems={[]}
        screenSize={{ breakpointPixels: 0, viewType: ViewType.Desktop }}
        sortBy=""
        totalResults={100}
        clearFilters={jest.fn()}
        setBrandFilters={jest.fn()}
        setCategoryFilters={jest.fn()}
        setSortBy={jest.fn()}
        setStoreFilters={jest.fn()}
        storeFilterItems={[]}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
