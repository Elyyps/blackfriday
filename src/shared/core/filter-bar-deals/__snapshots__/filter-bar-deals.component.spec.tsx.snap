// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FilterBarDeals] should render without crashing 1`] = `
<component
  brandFilterItems={Array []}
  categoryFilterItems={Array []}
  clearFilters={[MockFunction]}
  filtersChanged={[MockFunction]}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  screenSize={
    Object {
      "breakpointPixels": 0,
      "viewType": 3,
    }
  }
  setBrandFilters={[MockFunction]}
  setCategoryFilters={[MockFunction]}
  setSortBy={[MockFunction]}
  setStoreFilters={[MockFunction]}
  sortBy=""
  storeFilterItems={Array []}
  totalResults={100}
/>
`;
